python/lsst/analysis/tools/actions/plot/plotUtils.py

```diff
@@ -598,6 +598,68 @@ def addSummaryPlot(fig, loc, sumStats, label):
     return fig
 
 
+def shorten_list(numbers: Iterable[int]) -> str:
+    """Shorten a list of integers.
+
+    Parameters
+    ----------
+    numbers : `list` [`int`]
+        Any iterable (list, set, tuple) of integers.
+
+    Returns
+    -------
+    result : `str`
+        A shortened string representation of the list.
+
+    Examples
+    --------
+    >>> shorten_list([1,2,3,5,6,8])
+    "1-3,5-6,8"
+
+    >>> shorten_list((1,2,3,5,6,8,9,10,11))
+    "1-3,5-6,8-11"
+
+    >>> shorten_list(range(4))
+    "0-3"
+    """
+    if not numbers:  # empty container
```

Comment:
As we discussed:
```
if len(numbers) == 0:
```
will catch numpy array inputs.

Reply:
Alternatively, perhaps convert to a `set` right at the top of this function. This would then allow for the code block checking for possible duplicate number entires in your input iterable to be dropped.

Reply:
Option #3: you could swap the order of `len(numbers)` and `sorted` such that sorting occurs first. This will guarantee that your numbers iterable is in a list format, which would then have a `len`. Otherwise, the input (e.g., an iterator) may not have a `len` to pass the initial `if` check.

Reply:
I think I am going to go with option 3. If it is an empty container, calling `sorted` on it is a trivial operation. OTOH, creating a set out of a large container might be relatively more expensive. Option 1 is still the best for our use case, but I am going to refrain from assuming that __len__ is implemented for ``numbers``.